### PR TITLE
feat: Extend account onboarding template to include AppStream resources

### PR DESCRIPTION
**Pull request by jn1119**
Issue #, if available: GALI-914

Description of changes:

* Extended onboarding template to include AppStream Security Group, Fleet, and Stack.
* Also updated SageMaker endpoint to include a specific Security Group instead of using the default one. Note that this is required since we will only need SageMaker access for AppStream so same set of rules cannot apply across all the interface endpoints since all other endpoints will be accessed via workspaces.
* Added Security Group rules to ensure that AppStream instances can connect to SageMaker

Testing

* Verified the template with and without AppStream enabled in a member account
* Also verified that AppStream instances were able to connect via SageMaker. Note that I didn't verify EC2 access since that was already verified in the VPC setup PR

Checklist:

- [x] Have you successfully deployed to an AWS account with your changes?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully tested with your changes locally?
- [ ] If new dependencies have been added, have they been pinned to specific versions?
- [ ] Is this change also required on the AWS Solution version?
- [ ] Have you updated openapi.yaml if you made updates to API definition (including add, delete or update parameter and request data schema)?
- [ ] If you had to run manual tests, have you considered automating those tests by adding them to [end-to-end tests](../main/end-to-end-tests/README.md)?

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/onboard-account.cfn.yml [hunk 1]
```diff
@@ -552,7 +586,100 @@ Resources:
...
       VpcId: !Ref VPC
+      SecurityGroupEgress:
+        - CidrIp: 127.0.0.1/32
+          IpProtocol: '-1'
+
+  AppStreamSecurityGroupEgress:
+    Type: AWS::EC2::SecurityGroupEgress
+    Condition: isAppStream
+    Properties:
+      GroupId: !Ref AppStreamSecurityGroup
```
Comment (nguyen102) [inline addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/onboard-account.cfn.yml:619]: How does `GroupId` and `DestinationSecurityGroupId` work? I notice we reference different value for these two fields.

**Review by nguyen102 — COMMENTED** [review]
Overall looks good. Btw, why do we only set up SG for Sagemaker? Do we need SG for ec2, emr, and RStudio as well? 

The workspace we're launching are in the private VPCs that we created. Do we actually need SGs for them, since no resource outside the VPC can access these workspace anyways. Unless the reason for the SG is to prevent workspaces from talking to each other.

File: addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/onboard-account.cfn.yml (hunk 1 above)
Comment (jn1119) [reply addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/onboard-account.cfn.yml:619]: We are setting egress from `GroupId` to `DestinationSecurityGroupId`. Note that this is a different `Type` than standard SecurityGroup. We are allowing AppStream to connect with SageMaker

**Comment by jn1119** [comment]
> Overall looks good. Btw, why do we only set up SG for Sagemaker? Do we need SG for ec2, emr, and RStudio as well?
> 
> The workspace we're launching are in the private VPCs that we created. Do we actually need SGs for them, since no resource outside the VPC can access these workspace anyways. Unless the reason for the SG is to prevent workspaces from talking to each other.

I still need to verify the path for EMR and RStudio but each EC2 instance would have its own security group and the connectivity from AppStream instances won't be via interface endpoint- it would be via SSH. At that point EC2 instance security group and AppStream security group would need to go through similar egress/ingress setup. 

We don't want our workspaces unnecessary access to SageMaker interface endpoint. That can potentially mean that they have outside access to some arbitrary SageMaker notebook that we don't know about. Therefore, we have a different SG for SageMaker so that we can only give AppStream instances the access.

**Review by nguyen102 — APPROVED** [review]

**Pull request merged**